packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,174 @@
+      const repackSpawnAsync = createSpawnAsyncStepAdapter(verbose, stepsCtx.logger);
+
+      const tmpDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), `repack-`));
+      const workingDirectory = path.join(tmpDir, 'working-directory');
+      await fs.promises.mkdir(workingDirectory, { recursive: true });
+      stepsCtx.logger.info(`Created temporary workingDirectory: ${workingDirectory}`);
+
+      const sourceAppPath = inputs.source_app_path.value as string;
+      const outputPath =
+        (inputs.output_path.value as string) ??
+        path.join(tmpDir, `repacked${path.extname(sourceAppPath)}`);
+
+      stepsCtx.logger.info('Repacking the app...');
+      if (platform === Platform.IOS) {
+        await repackAppIosAsync({
+          platform: 'ios',
+          projectRoot,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add iosSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env: {
+            FASTLANE_DISABLE_COLORS: '1',
+            FASTLANE_SKIP_UPDATE_CHECK: '1',
+            SKIP_SLOW_FASTLANE_WARNING: 'true',
+            FASTLANE_HIDE_TIMESTAMP: 'true',
+            LC_ALL: 'en_US.UTF-8',
+            ...env,
+          },
+        });
+      } else if (platform === Platform.ANDROID) {
+        await repackAppAndroidAsync({
+          platform: 'android',
+          projectRoot,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add androidSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env,
+        });
+      } else {
+        throw new Error('Unsupported platform');
+      }
+
+      stepsCtx.logger.info(`Repacked the app to ${outputPath}`);
+      outputs.output_path.set(outputPath);
+    },
+  });
+}
+
+/**
+ * Creates a Bunyan logger adapter for repack logger
+ */
+export function createBunyanLoggerAdapter(logger: bunyan): Logger {
+  const timerMap: Record<string, number> = {};
+
+  return {
+    debug(...message: any[]): void {
+      logger.debug(message.join(''));
+    },
+    info(...message: any[]): void {
+      logger.info(message.join(''));
+    },
+    warn(...message: any[]): void {
+      logger.warn(message.join(''));
+    },
+    error(...message: any[]): void {
+      logger.error(message.join(''));
+    },
+    time(label: string): void {
+      timerMap[label] = Date.now();
+    },
+    timeEnd(label: string): void {
+      if (timerMap[label] == null) {
+        logger.warn(`Timer '${label}' does not exist`);
+        return;
+      }
+      const duration = Date.now() - timerMap[label];
+      logger.info(`${label}: ${duration} ms`);
+      delete timerMap[label];
+    },
+  };
+}
+
+/**
+ * Creates `@expo/steps` based spawnAsync for repack.
+ */
+function createSpawnAsyncStepAdapter(verbose: boolean, logger: bunyan): SpawnProcessAsync {
```

Comment:
```suggestion
function createSpawnAsyncStepAdapter({ verbose, logger }: { verbose: boolean, logger: bunyan }): SpawnProcessAsync {
```
I find these to be much more readable.